WikiWatershed/model-my-watershed
# Add analysis for drainage areas

## Overview

This takes an input AoI, finds its centroid, finds a HUC-12 that contains the centroid, and runs GWLF-E on it. Ultimately, we will scale the GWLF-E results to the input AoI, but for that we need the Drexel / ANS API. So this is not complete until that API has been integrated.

Connects #3553 

## Testing Instructions

* Check out this branch
* Go to http://localhost:8000/ and draw any drainage area
* Download the resulting AoI as GeoJSON
    <img width="1512" alt="image" src="https://user-images.githubusercontent.com/1430060/180518123-9f74a881-4f60-437e-a5c5-2e33c788d30f.png">
* Copy your API token
* Exercise the new API, like so:
    ```console
    http :8000/api/analyze/drainage-area/ Authorization:"Token $YOUR_TOKEN" < $DOWNLOADED_FILE.geojson
    ```
  - [x] Ensure the call succeeds
  - [x] Ensure the job results are GWLF-E outputs of Loads and monthlies
